network/a10/a10_server_axapi3.py

```diff
@@ -0,0 +1,257 @@
+      - port-number: 8443
+        protocol: TCP
+
+'''
+
+VALID_PORT_FIELDS = ['port-number', 'protocol', 'action']
+
+def validate_ports(module, ports):
+    for item in ports:
+        for key in item:
+            if key not in VALID_PORT_FIELDS:
+                module.fail_json(msg="invalid port field (%s), must be one of: %s" % (key, ','.join(VALID_PORT_FIELDS)))
+
+        # validate the port number is present and an integer
+        if 'port-number' in item:
+            try:
+                item['port-number'] = int(item['port-number'])
+            except:
+                module.fail_json(msg="port-number entries in the port definitions must be integers")
+        else:
+            module.fail_json(msg="port definitions must define the port-number field")
+
+        # validate the port protocol is present, no need to convert to the internal API integer value in v3
+        if 'protocol' in item:
+            protocol = item['protocol']
+            if not protocol:
+                module.fail_json(msg="invalid port protocol, must be one of: %s" % ','.join(AXAPI_PORT_PROTOCOLS))
+            else:
+                item['protocol'] = protocol
+        else:
+            module.fail_json(msg="port definitions must define the port protocol (%s)" % ','.join(AXAPI_PORT_PROTOCOLS))
+
+        # 'status' is 'action' in AXAPIv3
+        # no need to convert the status, a.k.a action, to the internal API integer value in v3
+        # action is either enabled or disabled
+        if 'action' in item:
+            action = item['action']
+            if action not in ['enable', 'disable']:
+                module.fail_json(msg="server action must be enable or disable")
+        else:
+            item['action'] = 'enable'
+
+
+def main():
+    argument_spec = a10_argument_spec()
+    argument_spec.update(url_argument_spec())
+    argument_spec.update(
+        dict(
+            operation=dict(type='str', default='create', choices=['create', 'update', 'delete']),
+            server_name=dict(type='str', aliases=['server'], required=True),
+            server_ip=dict(type='str', aliases=['ip', 'address'], required=True),
+            server_status=dict(type='str', default='enable', aliases=['action'], choices=['enable', 'disable']),
+            server_ports=dict(type='list', aliases=['port'], default=[]),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    host = module.params['host']
+    username = module.params['username']
+    password = module.params['password']
+    operation = module.params['operation']
+    write_config = module.params['write_config']
+    slb_server = module.params['server_name']
+    slb_server_ip = module.params['server_ip']
+    slb_server_status = module.params['server_status']
+    slb_server_ports = module.params['server_ports']
+
+    if slb_server is None:
+        module.fail_json(msg='server_name is required')
```

Comment:
Is this (and the `slp_server_ip` check a few lines down) actually needed you've set `required=True` in `argument_spec.update`, which I believe should do this for you